Flaky test VertxHttpClientAsyncBodyTest.consumeBytesNotProcessedIfCancelled

```
Error:  Tests run: 3, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.057 s <<< FAILURE! -- in io.fabric8.kubernetes.client.vertx.VertxHttpClientAsyncBodyTest
Error:  io.fabric8.kubernetes.client.vertx.VertxHttpClientAsyncBodyTest.consumeBytesNotProcessedIfCancelled -- Time elapsed: 0.015 s <<< FAILURE!
java.lang.AssertionError: 

Expecting actual throwable to be an instance of:
  java.util.concurrent.CancellationException
but was:
  java.util.concurrent.ExecutionException: io.vertx.core.http.StreamResetException: Stream reset: 0
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
	at io.fabric8.kubernetes.client.http.AbstractAsyncBodyTest.lambda$consumeBytesNotProcessedIfCancelled$2(AbstractAsyncBodyTest.java:104)
	...(83 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
	at io.fabric8.kubernetes.client.http.AbstractAsyncBodyTest.consumeBytesNotProcessedIfCancelled(AbstractAsyncBodyTest.java:105)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,7 +3,7 @@
 ### 7.7-SNAPSHOT
 
 #### Bugs
-* Fix #7694: (httpclient-vertx) clear response exception handler before reset in cancel() to prevent StreamResetException from racing with future cancellation
+* Fix #7696: (httpclient-vertx) clear response exception handler before reset in cancel() to prevent StreamResetException from racing with future cancellation
 * Fix #7632: java-generator now HTML-escapes `<`, `>`, and `&` in CRD descriptions to produce valid Javadoc
 * Fix #7543: fix processInlineDuplicateFields to recursively resolve nested inline embeds
 * Fix #7450: StandardHttpClient.shouldRetry() does not retry on Vert.x HttpClosedException

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -3,6 +3,7 @@
 ### 7.7-SNAPSHOT
 
 #### Bugs
+* Fix #7694: (httpclient-vertx) clear response exception handler before reset in cancel() to prevent StreamResetException from racing with future cancellation
 * Fix #7632: java-generator now HTML-escapes `<`, `>`, and `&` in CRD descriptions to produce valid Javadoc
 * Fix #7543: fix processInlineDuplicateFields to recursively resolve nested inline embeds
 * Fix #7450: StandardHttpClient.shouldRetry() does not retry on Vert.x HttpClosedException

--- a/httpclient-vertx/src/main/java/io/fabric8/kubernetes/client/vertx/VertxHttpRequest.java
+++ b/httpclient-vertx/src/main/java/io/fabric8/kubernetes/client/vertx/VertxHttpRequest.java
@@ -69,8 +69,12 @@ class VertxHttpRequest {
 
         @Override
         public void cancel() {
+          // The exception handler must be cleared before calling reset(), otherwise
+          // the reset triggers a StreamResetException that completes the future
+          // exceptionally before done.cancel() can run.
           resp.handler(null);
           resp.endHandler(null);
+          resp.exceptionHandler(null);
           resp.request().reset();
           done.cancel(false);
         }

